packages/playwright-core/src/client/browserContext.ts

```diff
@@ -432,6 +432,7 @@ export class BrowserContext extends ChannelOwner<channels.BrowserContextChannel>
       this._browser._contexts.delete(this);
     this._browserType?._contexts?.delete(this);
     this._disposeHarRouters();
+    this.tracing._resetStackCounterIfNeeded();
```

Comment:
Do the same for APIRequestContext?

Reply:
`page.request._tracing === context.tracing` - its the same Tracing instance under the hood.

Reply:
There is also a global `APIRequestContext` not attached to a `BrowserContext`.